### ISSUE TITLE
Failed building wheel for distribute when using debinate package 

### ISSUE THREAD

**Issue by whwright** [rholder/debinate#20]
I have seen this when building a few different debinate projects (all have been python3)

```
↪ export DEBINATE_PYTHON=$(which python3)
↪ debinate package
Cleaning out .debinate/target
Cleaning out .debinate/build
Building mything - 1.0.0
Clearing any virtualenv in /opt/mything/.virtualenv
Found requirements.txt, installing dependencies in order first...
Creating new virtualenv in /opt/mything/.virtualenv...
Using base prefix '/usr'
New python executable in /opt/mything/.virtualenv/bin/python3
Also creating executable in /opt/mything/.virtualenv/bin/python
Installing setuptools, pip, wheel...done.
Running virtualenv with interpreter /usr/bin/python3
Collecting distribute==0.7.3
  Using cached distribute-0.7.3.zip
Requirement already satisfied: setuptools>=0.7 in /opt/mything/.virtualenv/lib/python3.5/site-packages (from distribute==0.7.3)
Requirement already satisfied: appdirs>=1.4.0 in /opt/mything/.virtualenv/lib/python3.5/site-packages (from setuptools>=0.7->distribute==0.7.3)
Requirement already satisfied: packaging>=16.8 in /opt/mything/.virtualenv/lib/python3.5/site-packages (from setuptools>=0.7->distribute==0.7.3)
Requirement already satisfied: six>=1.10.0 in /opt/mything/.virtualenv/lib/python3.5/site-packages (from setuptools>=0.7->distribute==0.7.3)
Requirement already satisfied: pyparsing in /opt/mything/.virtualenv/lib/python3.5/site-packages (from packaging>=16.8->setuptools>=0.7->distribute==0.7.3)
Building wheels for collected packages: distribute
  Running setup.py bdist_wheel for distribute: started
  Running setup.py bdist_wheel for distribute: finished with status 'error'
  Complete output from command /opt/mything/.virtualenv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-zfa2fsu2/distribute/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp6oxttpbvpip-wheel- --python-tag cp35:
  Traceback (most recent call last):
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2680, in _dep_map
      return self.__dep_map
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2525, in __getattr__
      raise AttributeError(attr)
  AttributeError: _DistInfoDistribution__dep_map
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-build-zfa2fsu2/distribute/setup.py", line 58, in <module>
      setuptools.setup(**setup_params)
    File "/usr/lib/python3.5/distutils/core.py", line 134, in setup
      ok = dist.parse_command_line()
    File "/tmp/pip-build-zfa2fsu2/distribute/setuptools/dist.py", line 276, in parse_command_line
      result = _Distribution.parse_command_line(self)
    File "/usr/lib/python3.5/distutils/dist.py", line 472, in parse_command_line
      args = self._parse_command_opts(parser, args)
    File "/tmp/pip-build-zfa2fsu2/distribute/setuptools/dist.py", line 602, in _parse_command_opts
      nargs = _Distribution._parse_command_opts(self, parser, args)
    File "/usr/lib/python3.5/distutils/dist.py", line 528, in _parse_command_opts
      cmd_class = self.get_command_class(command)
    File "/tmp/pip-build-zfa2fsu2/distribute/setuptools/dist.py", line 406, in get_command_class
      ep.require(installer=self.fetch_build_egg)
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2254, in require
      working_set.resolve(self.dist.requires(self.extras),env,installer)))
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2471, in requires
      dm = self._dep_map
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2682, in _dep_map
      self.__dep_map = self._compute_dependencies()
    File "/tmp/pip-build-zfa2fsu2/distribute/pkg_resources.py", line 2699, in _compute_dependencies
      from _markerlib import compile as compile_marker
  ImportError: No module named '_markerlib'
  
  ----------------------------------------
  Failed building wheel for distribute
  Running setup.py clean for distribute
Failed to build distribute
Installing collected packages: distribute
  Running setup.py install for distribute: started
    Running setup.py install for distribute: finished with status 'done'
Successfully installed distribute-0.7.3
...
```

and it continues on to install all dependencies defined in `requirements.txt`. `debinate package` goes on to finish, and I have not found any issues with the generated deb files when this distribute error is present.

I have also found that if I remove my `requirements.txt` file, and instead define the dependencies in `setup.py` debinate follows [this code path](https://github.com/rholder/debinate/blob/master/debinate#L135) instead where `pip install distribute==0.7.3` is not run. I do not see the error in this case, and I have also not experienced issues with the generated deb file this way either.

A little more details on my environment:

```
↪ cat /etc/os-release                                                                                                                                                                            1 ↵
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.1 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

↪ debinate --version
Debinate 0.6.0 - roll up your project into a Debian package

↪ python3 --version                                                                                                                                                                              1 ↵
Python 3.5.2
```

TL;DR I don't know if this is an issue or not (hasn't been for me), but a distribute error in the logs is a little worrisome.

**Comment by rholder** [comment]
I agree. I don't like that error in the logs. Per my cryptic TODO just above the section you highlighted, maybe we don't need to explicitly install `distribute` at all anymore. I recall it only ever being an issue with an older version of `MySQL-python`.

**Comment by wright8191** [comment]
#2 says 

> The version is in there now to get around annoying MySQL-python minimum version dependencies.

**Comment by whwright** [comment]
I got a new error when building a `python3.6` project:

```
Running virtualenv with interpreter /usr/local/bin/python3.6
Using base prefix '/usr/local'
New python executable in /opt/vandelay-cron/.virtualenv/bin/python3.6
Also creating executable in /opt/vandelay-cron/.virtualenv/bin/python
Installing setuptools, pip, wheel...done.
Collecting distribute==0.7.3
  Using cached distribute-0.7.3.zip
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-rk9myknr/distribute/setuptools/__init__.py", line 2, in <module>
        from setuptools.extension import Extension, Library
      File "/tmp/pip-build-rk9myknr/distribute/setuptools/extension.py", line 5, in <module>
        from setuptools.dist import _get_unpatched
      File "/tmp/pip-build-rk9myknr/distribute/setuptools/dist.py", line 7, in <module>
        from setuptools.command.install import install
      File "/tmp/pip-build-rk9myknr/distribute/setuptools/command/__init__.py", line 8, in <module>
        from setuptools.command import install_scripts
      File "/tmp/pip-build-rk9myknr/distribute/setuptools/command/install_scripts.py", line 3, in <module>
        from pkg_resources import Distribution, PathMetadata, ensure_directory
      File "/tmp/pip-build-rk9myknr/distribute/pkg_resources.py", line 1518, in <module>
        register_loader_type(importlib_bootstrap.SourceFileLoader, DefaultProvider)
    AttributeError: module 'importlib._bootstrap' has no attribute 'SourceFileLoader'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-rk9myknr/distribute/
```

I removed the `pip install distribute==0.7.3` line and things work fine

**Comment by rholder** [comment]
This should now be resolved as of #23 going in and is now part of the latest 0.7.0 release.
